app/main/helpers/framework_helpers.py

```diff
@@ -63,3 +64,20 @@ def get_framework_or_500(client, framework_slug, logger=None):
             abort(500, f'Framework not found: {framework_slug}')
         else:
             raise
+
+
+def get_following_framework_or_500(client, current_framework, logger):
+    try:
+        following_framework_content = content_loader.get_metadata(
+            current_framework['slug'], 'following_framework', 'framework'
+        )
+        if following_framework_content:
+            # We raise an error if the following_framework content refers to a framework not in the database
+            return get_framework_or_500(
+                client,
+                following_framework_content['slug'],
```

Comment:
I presume this means that putting G10 in a pre-live state will cause a 500? Do we care about that?

Reply:
Hmm that's a good point - we should fail silently instead in case production isn't ready to show the message.

Reply:
Hopefully we should never get to that state in production? Do we have functional tests that would catch a failure?

Reply:
Well, we only want to create the prod db entry for G11 when we're ready to show the 'coming' message. If we deploy the content update before then, the view will find the `following_framework` content and will be expecting the data to go with it.

Reply:
Ah I see it used to fail silently as well. Fine then 👍